doc/kernel/services/data_passing/pipes.rst

```diff
@@ -203,17 +203,11 @@ data in the pipe.
 Suggested uses
 **************
 
-.. code-block:: c
+Use a pipe to send streams of data between threads.
+
+.. note::
+    A pipe can be used to transfer long streams of data if desired. However it is often preferable to send pointers to large data items to avoid copying the data.
 
-    void monitor_thread(void)
-    {
-        while (1) {
-            ...
-            /* Critical error detected. Flush the entire pipe to reset it. */
-            k_pipe_flush(&my_pipe);
-            ...
-        }
-    }
```

Comment:
I think you had good catch in observing that code block is under the "Suggested uses" section. However, removing the code block is not the right thing to do. Instead it should simply be moved up above the "Suggested uses" so that it remains associated with the "Flushing a Pipe's Buffer" section.

Reply:
That is true, I'll update with this change and the commit message fix